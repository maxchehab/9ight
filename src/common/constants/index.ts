export { RequestMethod } from './request-method.enum';
