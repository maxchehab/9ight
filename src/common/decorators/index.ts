export {
  Connect,
  Delete,
  Get,
  Head,
  Options,
  Patch,
  Post,
  Put,
} from './request-mapping.decorators';
