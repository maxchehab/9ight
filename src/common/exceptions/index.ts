export * from './bad-gateway.exception';
export * from './http.exception';
export * from './network-connect-timeout-error.exception';
export * from './request-timeout.exception';
export * from './bad-request.exception';
export * from './im-a-teapot.exception';
export * from './not-acceptable.exception';
export * from './request-uri-too-long.exception';
export * from './client-closed-request.exception';
export * from './not-extended.exception';
export * from './requested-range-not-satisfiable.exception';
export * from './conflict.exception';
export * from './insufficient-storage.exception';
export * from './not-found.exception';
export * from './service-unavailable.exception';
export * from './connection-closed-without-response.exception';
export * from './internal-server-error.exception';
export * from './not-implemented.exception';
export * from './too-many-requests.exception';
export * from './expectation-failed.exception';
export * from './length-required.exception';
export * from './payload-too-large.exception';
export * from './unauthorized.exception';
export * from './failed-dependency.exception';
export * from './locked.exception';
export * from './payment-required.exception';
export * from './unavailable-for-legal-reasons.exception';
export * from './forbidden.exception';
export * from './loop-detected.exception';
export * from './precondition-failed.exception';
export * from './unprocessable-entity.exception';
export * from './gateway-timeout.exception';
export * from './method-not-allowed.exception';
export * from './precondition-required.exception';
export * from './unsupported-media-type.exception';
export * from './gone.exception';
export * from './misdirected-request.exception';
export * from './proxy-authentication-required.exception';
export * from './upgrade-required.exception';
export * from './http-version-not-supported.exception';
export * from './network-authentication-required.exception';
export * from './request-header-fields-too-large.exception';
export * from './variant-also-negotiates.exception';
