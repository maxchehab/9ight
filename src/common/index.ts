export * from './constants';
export * from './decorators';
export * from './exceptions';
export * from './interfaces';
export * from './utils';
