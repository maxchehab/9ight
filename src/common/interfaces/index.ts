export { ClassType } from './class-type.interface';
export { DecoratorTarget } from './decorator-target.interface';
export { Parameter, ParameterTransformer } from './parameter.interface';
