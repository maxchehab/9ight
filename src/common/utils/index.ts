export * from './generate-url-from-methods.util';
export * from './find-method-and-params.util';
