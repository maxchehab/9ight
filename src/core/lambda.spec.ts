import { Lambda } from '.';
import { Get } from '../common';

describe('Lambda', () => {
  it('throws a no-method-decorators error when a class does not have any method decorators', async () => {
    class Users {
      list(): string[] {
        return ['all', 'of', 'the', 'users'];
      }
    }

    expect(() => Lambda(Users)).toThrowErrorMatchingSnapshot();
  });

  it('does not throw a no-method-decorators error when a class contains a method decorator', async () => {
    class Users {
      @Get()
      list(): string[] {
        return ['all', 'of', 'the', 'users'];
      }
    }

    expect(() => Lambda(Users)).not.toThrow();
  });
});
