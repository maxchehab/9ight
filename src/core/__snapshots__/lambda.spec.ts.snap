// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Lambda throws a no-method-decorators error when a class does not have any method decorators 1`] = `"The class Users does not have any method decorators https://err.sh/maxchehab/9ight/no-method-decorators-found."`;
